fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -43,33 +43,53 @@ impl<T> Interpreter<PredicateStorage, T> {
     ///
     /// This is not a valid entrypoint for debug calls. It will only return a `bool`, and not the
     /// VM state required to trace the execution steps.
-    pub fn check_predicates<Tx>(checked: Checked<Tx>, params: ConsensusParameters, gas_costs: GasCosts) -> bool
+    pub fn check_predicates<Tx>(
+        checked: Checked<Tx>,
+        params: ConsensusParameters,
+        gas_costs: GasCosts,
+    ) -> Result<PredicatesChecked, PredicateVerificationFailed>
     where
-        Tx: ExecutableTransaction,
+        Tx: ExecutableTransaction + fuel_tx::field::GasLimit,
```

Comment:
Hmm, how that worked before without `fuel_tx::field::GasLimit`? I guess you need functionality from `Chargeable` trait.